### PR TITLE
feat: Onboard Census Bureau International Dataset

### PR DESCRIPTION
**Pull request by vasuc-google**
## Description

Based on #<ISSUE-NUMBER>

Note: It's recommended to open an issue first for context and discussion.

## Checklist

Note: Delete items below that aren't applicable to your pull request.

- [x] Please merge this PR for me once it is approved.

- [x] If this PR adds or edits a dataset or pipeline, it was reviewed and approved by the Google Cloud Public Datasets team beforehand.
- [x] If this PR adds or edits a dataset or pipeline, I put all my code inside  `datasets/census_bureau_international` and nothing outside of that directory.

- [x] This PR is appropriately labeled.

### CODE REVIEW

**Comment by adlersantos** [comment]
@vasuc-google Kindly address the Python linting errors and we should be good to go

File: datasets/census_bureau_international/_images/run_csv_transform_kub/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The base image for this build
+# FROM gcr.io/google.com/cloudsdktool/cloud-sdk:352.0.0-slim
+# FROM python:3.8
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/_images/run_csv_transform_kub/Dockerfile:17]: Remove these commented lines

File: datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        df = df
+    else:
+        logging.info("Filter Headers ...")
+        df = df.drop(["country_area"], axis=1)
+
+    logging.info("Search and Replacing the values..")
+    if pipeline_name == "midyear_population_age_sex":
+        df["sex"] = df["sex"].apply({2: "Male", 3: "Female"}.get)
+    else:
+        df = df
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py:60]: Remove the else part . Not required

File: datasets/census_bureau_international/_images/run_csv_transform_kub/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+requests
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/_images/run_csv_transform_kub/requirements.txt:1]: Remove request as you are pulling data using gsutil...

File: datasets/census_bureau_international/country_names_area/pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    description: "Country_Names_Area Dataset"
+
+dag:
+  airflow_version: 1
+  initialize:
+    dag_id: "country_names_area"
+    default_args:
+      owner: "Google"
+
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/country_names_area/pipeline.yaml:29]: remove the space

File: datasets/census_bureau_international/country_names_area/pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      args:
+
+        task_id: "transform_csv"
+
+        startup_timeout_seconds: 600
+
+        name: "country_names_area"
+
+        namespace: "default"
+
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/country_names_area/pipeline.yaml:51]: remove spaces...

File: datasets/census_bureau_international/country_names_area/pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                - matchExpressions:
+                    - key: cloud.google.com/gke-nodepool
+                      operator: In
+                      values:
+                        - "pool-e2-standard-4"
+
+
+        image_pull_policy: "Always"
+
+        image: "{{ var.json.census_bureau_international.container_registry.run_csv_transform_kub }}"
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/country_names_area/pipeline.yaml:65]: ^

File: datasets/census_bureau_international/country_names_area/pipeline.yaml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        bucket: "{{ var.value.composer_bucket }}"
+
+        source_objects: ["data/census_bureau_international/country_names_area/data_output.csv"]
+        source_format: "CSV"
+        destination_project_dataset_table: "census_bureau_international.country_names_area"
+
+        skip_leading_rows: 1
+        allow_quoted_newlines: True
+
+        write_disposition: "WRITE_TRUNCATE"
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/country_names_area/pipeline.yaml:98]: ^

File: datasets/census_bureau_international/dataset.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+      Update frequency: Historic (none)
+
+      Dataset source: United States Census Bureau
+
+      Terms of use: This dataset is publicly available for anyone to use under the following terms provided by the Dataset Source - http://www.data.gov/privacy-policy#data_policy - and is provided "AS IS" without any warranty, express or implied, from Google. Google disclaims all liability for any damages, direct or indirect, resulting from the use of the dataset.
+
+      See the GCP Marketplace listing for more details and sample queries: https://console.cloud.google.com/marketplace/details/united-states-census-bureau/international-census-data
+
+  dataset_sources: ~
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/dataset.yaml:32]: ^

File: datasets/census_bureau_international/dataset.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    description: |-
+      Overview: The United States Census Bureau’s international dataset provides estimates of country populations since 1950 and projections through 2050. Specifically, the dataset includes midyear population figures broken down by age and gender assignment at birth. Additionally, time-series data is provided for attributes including fertility rates, birth rates, death rates, and migration rates.
+
+      Update frequency: Historic (none)
+
+      Dataset source: United States Census Bureau
+
+      Terms of use: This dataset is publicly available for anyone to use under the following terms provided by the Dataset Source - http://www.data.gov/privacy-policy#data_policy - and is provided "AS IS" without any warranty, express or implied, from Google. Google disclaims all liability for any damages, direct or indirect, resulting from the use of the dataset.
+
+      See the GCP Marketplace listing for more details and sample queries: https://console.cloud.google.com/marketplace/details/united-states-census-bureau/international-census-data
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/dataset.yaml:50]: ^

File: datasets/census_bureau_international/midyear_population/pipeline.yaml [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    schedule_interval: "@daily"
+    catchup: False
+    default_view: graph
+
+  tasks:
+    - operator: "KubernetesPodOperator"
+
+      description: "Run CSV transform within kubernetes pod"
+
+      args:
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/midyear_population/pipeline.yaml:42]: ^

File: datasets/census_bureau_international/midyear_population_5yr_age_sex/pipeline.yaml [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+      depends_on_past: False
+      start_date: '2021-03-01'
+    max_active_runs: 1
+    schedule_interval: "@daily"
+    catchup: False
+    default_view: graph
+
+  tasks:
+    - operator: "KubernetesPodOperator"
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/midyear_population_5yr_age_sex/pipeline.yaml:38]: ^

File: datasets/census_bureau_international/midyear_population_age_sex/pipeline.yaml [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+      owner: "Google"
+
+      depends_on_past: False
+      start_date: '2021-03-01'
+    max_active_runs: 1
+    schedule_interval: "@daily"
+    catchup: False
+    default_view: graph
+
+  tasks:
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/midyear_population_age_sex/pipeline.yaml:37]: ^

File: datasets/census_bureau_international/mortality_life_expectancy/pipeline.yaml [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+    default_args:
+      owner: "Google"
+
+      depends_on_past: False
+      start_date: '2021-03-01'
+    max_active_runs: 1
+    schedule_interval: "@daily"
+    catchup: False
+    default_view: graph
+
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/mortality_life_expectancy/pipeline.yaml:35]: ^

**Review by dipannitab2392 — CHANGES_REQUESTED** [review]

File: datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+    logging.info(f"Downloading file {source_url}")
+    download_file(source_url, source_file)
+
+    logging.info(f"Reading the file {source_url}")
+    if pipeline_name == "country_names_area":
+        df = pd.read_csv(source_file[0])
+    else:
+        df1 = pd.read_csv(source_file[0])
+        df2 = pd.read_csv(source_file[1])
+        df = pd.merge(df1, df2, how="left", on=["country_code"])
```
Comment (dipannitab2392) [inline datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py:48]: You can use below code : 

```suggestion
    def read_files(source_files, joining_key):
    df = pd.DataFrame()
    for source_file in source_files:
        _df = pd.read_csv(source_file)
        if df.empty:
            df = _df
        else:
            df = pd.merge(df, _df, how="left", on=["country_code"])
    return df
```

**Review by dipannitab2392 — CHANGES_REQUESTED** [review]

File: datasets/census_bureau_international/_images/run_csv_transform_kub/Dockerfile (hunk 1 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/_images/run_csv_transform_kub/Dockerfile]: done

File: datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py (hunk 2 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/_images/run_csv_transform_kub/csv_transform.py]: done

File: datasets/census_bureau_international/_images/run_csv_transform_kub/requirements.txt (hunk 3 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/_images/run_csv_transform_kub/requirements.txt]: done

File: datasets/census_bureau_international/country_names_area/pipeline.yaml (hunk 4 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/country_names_area/pipeline.yaml]: done

File: datasets/census_bureau_international/country_names_area/pipeline.yaml (hunk 5 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/country_names_area/pipeline.yaml]: done

File: datasets/census_bureau_international/country_names_area/pipeline.yaml (hunk 6 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/country_names_area/pipeline.yaml]: done

File: datasets/census_bureau_international/country_names_area/pipeline.yaml (hunk 7 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/country_names_area/pipeline.yaml]: done

File: datasets/census_bureau_international/dataset.yaml (hunk 8 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/dataset.yaml]: done

File: datasets/census_bureau_international/dataset.yaml (hunk 9 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/dataset.yaml]: done

File: datasets/census_bureau_international/midyear_population/pipeline.yaml (hunk 10 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/midyear_population/pipeline.yaml]: done

File: datasets/census_bureau_international/midyear_population_5yr_age_sex/pipeline.yaml (hunk 11 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/midyear_population_5yr_age_sex/pipeline.yaml]: done

File: datasets/census_bureau_international/midyear_population_age_sex/pipeline.yaml (hunk 12 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/midyear_population_age_sex/pipeline.yaml]: done

File: datasets/census_bureau_international/mortality_life_expectancy/pipeline.yaml (hunk 13 above)
Comment (vasuc-google) [reply datasets/census_bureau_international/mortality_life_expectancy/pipeline.yaml]: done

**Comment by dipannitab2392** [comment]
> @vasuc-google Kindly address the Python linting errors before I proceed with the review, thanks!
> 
> cc @dipannitab2392

@adlersantos - review is done from my side. You can have a look.

**Review by dipannitab2392 — COMMENTED** [review]
Review completed from my end

**Comment by nlarge-google** [comment]
closing this PR due to re-engineering in https://github.com/GoogleCloudPlatform/public-datasets-pipelines/pull/217/files

**Pull request closed without merging**